src/Data/Vec.agda

```diff
@@ -56,3 +59,20 @@ module _ {P : A → Set p} (P? : Decidable P) where
   dropWhile (a Vec.∷ as) with does (P? a)
   ... | true  = ≤-cast (ℕₚ.n≤1+n _) (dropWhile as)
   ... | false = fromVec (a Vec.∷ as)
+
+------------------------------------------------------------------------
+-- Padding a list of vectors
+
+rectangle : ∀[ Vec≤ A ⇒ Vec A ] → List (∃ (Vec A)) → ∃ (List ∘ Vec A)
+rectangle f = Prod.map₂ (List.map f)
+            ∘ Vec≤.rectangle
+            ∘ List.map (Prod.map₂ fromVec)
+
+rectangleˡ : A → List (∃ (Vec A)) → ∃ (List ∘ Vec A)
```

Comment:
I have to confess these function signatures are really doing my head in. They take a list of pairs `(n_1, Vec A n_1), ...,` and they presumably output a pair `(m, List (Vec A m))` where `m` is the maximum over `n_i`?

These seem like really specialised functions... I'm torn about whether or not they should live in `Data.Vec`? Actually are they even used in `Data.Vec.Show`? I can't seem to trace them through?

Reply:
They are not used. I wrote them before I wrote the `rectangle` ones for
`String` because I figured it was better for efficiency to work directly
on strings rather than to go via `Vec Char`. Their types however are more
precise and they could be used as specifications for the ones acting on
strings if we ever wanted to prove them equivalent!

Reply:
> They are not used.

Should we remove them for the moment then until we find a use for them?

I'm also a bit worried about adding more stuff to `Data.Vec` itself. Eventually after some of the issues with re-exporting are fixed,  in v2.0 I hope to have the main modules for every data type act as summary modules, re-exporting much more of the sub-hierarchy. If you'd like to keep them perhaps we could put them in a submodule, `Data.Vec.Reshape` maybe?

Reply:
Additionally there doesn't seem to much of a pattern where functions that involve both `Vec` and `Vec≤` live. I kind of feel that either they should all live in the `Vec` folder or in the `Vec≤` folder. Thoughts?